### PR TITLE
Implemented interface for bls_signatures

### PR DESCRIPTION
**Pull request by systemzax**
## Change Description

This pull request introduces aggregate signatures (BLS signature scheme) as a protocol feature. It uses Chia's BLS-signatures library, which in turn depends on the Relic toolkit.

Dependent on https://github.com/AntelopeIO/leap/pull/661

## API Changes

It defines the following new host functions : 

bls_verify
bls_aggregate_pubkeys
bls_aggregate_sigs
bls_aggregate_verify

as well as new built-in types :

bls_public_key
bls_signature

Unit tests are included in libraries/libfc/test/test_bls.cpp

## Documentation Additions

Documentation of the new features will be provided on demand

### CODE REVIEW

File: libraries/eosiolib/core/eosio/crypto_bls_ext.hpp [hunk 1]
```diff
@@ -0,0 +1,103 @@
...
+         int32_t bls_aggregate_sigs( const char* sigs, uint32_t sigs_len, const char* agg, uint32_t agg_len);
+
+         __attribute__((eosio_wasm_import))
+         bool bls_aggregate_verify( const char* sig, uint32_t sig_len, const char* digs, uint32_t digs_len, char* pubs, uint32_t pubs_len);
+
+      }
+
+   }
+
+   bool bls_verify(bls_public_key pub, bls_signature sig , std::vector<uint8_t> msg) {
```
Comment (dimas1185) [inline libraries/eosiolib/core/eosio/crypto_bls_ext.hpp:42]: pass by reference or explain why do you pass parameters by value here. same for other wrappers

File: libraries/eosiolib/core/eosio/crypto_bls_ext.hpp [hunk 2]
```diff
@@ -0,0 +1,103 @@
...
+
+      auto pub_data = eosio::pack(pubkeys);
+      auto sig_data = eosio::pack(sig);
+      auto msg_data = eosio::pack(messages);
+
+      return internal_use_do_not_use::bls_aggregate_verify( sig_data.data(), sig_data.size(), msg_data.data(), msg_data.size(), pub_data.data(), pub_data.size() );
+
+   }
+
+/*   bool bls_verify( const char* sig, uint32_t sig_len, const char* dig, uint32_t dig_len, char* pub, uint32_t pub_len) {
```
Comment (dimas1185) [inline libraries/eosiolib/core/eosio/crypto_bls_ext.hpp:99]: remove unnecessary comment

**Review by dimas1185 — CHANGES_REQUESTED** [review]
please add integration tests. see comment for similar PR: https://github.com/AntelopeIO/cdt/pull/154#pullrequestreview-1504500825

**Comment by arhag** [comment]
This was an old approach that we will no longer be going with. Instead we will use the existing BLS host functions merged into the `main` branch of Leap to implement verification in contract as needed.

**Pull request closed without merging**